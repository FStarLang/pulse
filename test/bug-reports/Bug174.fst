module Bug174

#lang-pulse

open Pulse
open FStar.SizeT

[@@expect_failure]
fn bad_range
  (#tt : Type0)
  (r : ref tt)
  (#v : erased tt)
  requires Pulse.Lib.Reference.pts_to r v
  ensures  Pulse.Lib.Reference.pts_to r v
{
  let v = Pulse.Lib.Reference.op_Bang #t r;
  admit();
}
