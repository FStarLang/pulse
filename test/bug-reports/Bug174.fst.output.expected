* Info at Bug174.fst(15,1-17,10):
  - Expected failure:
  - Ill-typed term: !r
  - Assertion failed
  - The SMT solver could not prove the query. Use --query_stats for more
    details.
  - Also see: Bug174.fst(16,30-16,42)
  - Other related locations: Bug174.fst(16,39-16,40)

